src/SfxWeb/src/app/views/cluster/federation/federation.component.ts

```diff
@@ -0,0 +1,656 @@
+import { Component, Injector, OnDestroy, OnInit } from '@angular/core';
+import { Observable, of, Subject, Subscription } from 'rxjs';
+import { debounceTime, distinctUntilChanged } from 'rxjs/operators';
+import { IResponseMessageHandler } from 'src/app/Common/ResponseMessageHandlers';
+import { NodeEventList, PartitionEventList } from 'src/app/Models/DataModels/collections/Collections';
+import { NodeEvent, PartitionEvent } from 'src/app/Models/eventstore/Events';
+import { IOnDateChange } from 'src/app/modules/event-store/double-slider/double-slider.component';
+import { IEventStoreData, IQuickDates } from 'src/app/modules/event-store/event-store/event-store.component';
+import { IOptionConfig } from 'src/app/modules/event-store/option-picker/option-picker.component';
+import { DataService } from 'src/app/services/data.service';
+import { TelemetryService } from 'src/app/services/telemetry.service';
+import { TimeUtils } from 'src/app/Utils/TimeUtils';
+import { BaseControllerDirective } from 'src/app/ViewModels/BaseController';
+import { Network} from "vis-network/standalone"
```

Comment:
The most significant changed here would be reworking the structure to take advantage of angular data bindings and template bindings to not access elements through getElementById and not set innerHTML/build HTML in the function to set the dom view. These can make the code more complicated and lose the advantage angular provides for keeping the state in the component in sync with the view.

Reply:
Thank you very much for detailed review, I updated most of the things so you can take a look and see if there is still something to repair